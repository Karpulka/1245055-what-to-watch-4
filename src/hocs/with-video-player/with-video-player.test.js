import React from "react";
import renderer from "react-test-renderer";
import withVideoPlayer from "./with-video-player";
import PropTypes from "prop-types";

const MockComponent = (props) => {
  return <div>{props.children}</div>;
};

MockComponent.propTypes = {
  children: PropTypes.element.isRequired
};

const src = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
const poster = `/poster.jpg`;

it(`Render withAudioPlayer`, () => {
  const VideoPlayerWrapper = withVideoPlayer(MockComponent);

  const tree = renderer
    .create(<VideoPlayerWrapper
      src={src}
      poster={poster}
      isStartPlaying={false}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
