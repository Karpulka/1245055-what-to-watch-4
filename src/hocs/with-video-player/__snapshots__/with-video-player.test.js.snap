// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withAudioPlayer 1`] = `
<div>
  <video />
</div>
`;
